tests/integration/helpers.py

```diff
@@ -485,8 +528,133 @@ def get_hostname(ops_test: OpsTest, unit_name: str) -> str:
+        "headers": json.dumps(headers),
+        "method": method,
+        "endpoint": endpoint,
+    }
+    if payload:
+        params["payload"] = payload
+    if server_uri:
+        params["server-uri"] = server_uri
+
+    action = await ops_test.model.units.get(unit_name).run_action(action, **(params or {}))
+    action = await action.wait()
+
+    if action.status != "completed":
+        raise Exception(action.results)
+
+    return action.results
+
+
+@retry(wait=wait_fixed(wait=15), stop=stop_after_attempt(15))
+async def client_run_db_request(
+    ops_test,
+    unit_name: str,
+    relation: Relation,
+    method: str,
+    endpoint: str,
+    payload: str = None,
+):
+    """Client applicatoin issuing a request to Opensearch."""
+    headers = {"Content-Type": "application/json", "Accept": "application/json"}
+    return await client_run_request(
+        ops_test, unit_name, relation, method, headers, endpoint, payload
+    )
+
+
+@retry(wait=wait_fixed(wait=15), stop=stop_after_attempt(15))
+async def client_run_dashboards_request(
+    ops_test,
+    unit_name: str,
+    relation: Relation,
+    method: str,
+    host: str,
+    endpoint: str,
+    payload: str = None,
+    https=False,
+):
+    """Client applicatoin issuing a request to Opensearch Dashboards."""
+    action = "run-dashboards-request"
+    headers = {
+        "Accept": "application/json",
+        "Content-Type": "application/json",
+        "osd-xsrf": "osd-fetch",
+    }
+    proto = "https" if https else "http"
+    server_uri = f"{proto}://{host}:5601"
+    return await client_run_request(
+        ops_test,
+        unit_name,
+        relation,
+        method,
+        headers,
+        endpoint,
+        payload,
+        action,
+        server_uri,
+    )
+
+
+async def client_run_all_dashboards_request(
+    ops_test: OpsTest,
+    unit_name: str,
+    relation: Relation,
+    method: str,
+    endpoint: str,
+    payload: str = None,
+    https: bool = False,
+):
+    """Check if all dashboard instances are accessible."""
+
+    result = []
+    if not ops_test.model.applications[APP_NAME].units:
+        logger.debug(f"No units for application {APP_NAME}")
+        return False
+
+    for dashboards_unit in ops_test.model.applications[APP_NAME].units:
+        host = get_private_address(ops_test.model.name, dashboards_unit.name)
+        if not host:
+            logger.debug(f"No hostname found for {dashboards_unit.name}, can't check connection.")
+            return False
+
+        response = await client_run_dashboards_request(
+            ops_test, unit_name, relation, method, host, endpoint, payload, https
+        )
+        if "results" in response:
+            result.append(json.loads(response["results"])["rawResponse"])
+        else:
+            result.append(response)
+        logger.info(f"Response from {unit_name}, {host}: {response}")
+
+    if result:
+        assert all(item == result[0] for item in result)
```

Comment:
Could you explain this? what happens if `result[0]` return an error response?

Reply:
I would like to get the same response from all dashboards.
I'm comparing the rest against the results from the 1st one. A good way to check for full equality, as at this point we can assume that `resultsl` has at least one member.

Reply:
However, true that since I do compare all results from all dsahboads this check can go away.
(Not to have discepency if ther may be timestaps or query duration in the response).